parsers/US_PJM.py

```diff
@@ -65,332 +72,283 @@
-    load = re.search(load_pattern, str(exchange_script)).group(1)
-    load_vals = demjson.decode(load)[0]
-
-    # Occasionally load_vals contains a null at the end of the list which must be caught.
-    actual_load = [float(val) for val in load_vals if val is not None]
-
+    # x-axis (time)
     time_pattern = r"var timeArray = (\[(.*)\])"
     time_array = re.search(time_pattern, str(exchange_script)).group(1)
     time_vals = demjson.decode(time_array)
 
-    flows = zip(actual_load, time_vals, strict=True)
-
-    date = datetime.combine(datetime.now(TIMEZONE), time())  # truncate to day
+    # y-axis [right] (actual & scheduled load)
+    load_pattern = r"var load = (\[(.*)\])"
+    load = re.search(load_pattern, str(exchange_script)).group(1)
+    load_actual = demjson.decode(load)[0]
 
     converted_flows = []
-    for flow in flows:
-        time_of_the_day = datetime.strptime(
-            flow[1],
-            "%I:%M %p",  # make sure to use %I and not %H for %p to take effect
-        ).replace(tzinfo=TIMEZONE)
-        dt = date.replace(hour=time_of_the_day.hour, minute=time_of_the_day.minute)
-        converted_flow = (flow[0], dt)
-        converted_flows.append(converted_flow)
-
-    return converted_flows
-
-
-def combine_NY_exchanges(session: Session) -> list:
-    """
-    Combination function for the 4 New York interfaces.
-    Timestamps are checked to ensure correct combination.
-    """
-
-    nyiso = get_exchange_data("nyiso", session)
-    neptune = get_exchange_data("neptune", session)
-    linden = get_exchange_data("linden", session)
-    hudson = get_exchange_data("hudson", session)
-
-    combined_flows = zip(nyiso, neptune, linden, hudson, strict=True)
-
-    flows = []
-    for datapoint in combined_flows:
-        total = sum([n[0] for n in datapoint])
-        stamps = [n[1] for n in datapoint]
-
-        # Data quality check to make sure timestamps all match.
-        if len(set(stamps)) == 1:
-            dt = stamps[0]
-        else:
-            # Drop bad datapoint and move to next.
+    today = datetime.combine(datetime.now(TIMEZONE), time(), tzinfo=TIMEZONE)
+    for t, flow in zip(time_vals, load_actual, strict=True):
+        # some tail values might be null
+        if flow is None:
             continue
 
-        flows.append((total, dt))
+        # make sure to use %I and not %H for %p to take effect
+        time_of_the_day = datetime.strptime(t, "%I:%M %p").replace(tzinfo=TIMEZONE)
+        dt = today.replace(hour=time_of_the_day.hour, minute=time_of_the_day.minute)
 
-    return flows
+        converted_flows.append((dt, float(flow)))
+
+    return converted_flows
 
 
 def fetch_exchange(
-    zone_key1: str,
-    zone_key2: str,
-    session: Session = Session(),
+    zone_key1: ZoneKey,
+    zone_key2: ZoneKey,
+    session: Session | None = None,
     target_datetime: datetime | None = None,
     logger: Logger = getLogger(__name__),
-) -> list[dict] | dict:
+) -> list[dict]:
```

Comment:
note: although this parses is not used anywhere at the moment to fetch exchanges, the previous function was broken and using zone keys that do not exist. Now it's fixed and the logic has also been expanded to support all possible exchanges